### PR TITLE
Fix Javadocs issues in the topic-operator module

### LINKED ISSUE #7703: Fix Javadocs in the `topic-operator` module
By default we configure the Maven Javadoc plugin to raise any warnings as errors when validating the Javadocs in our code. That helps us to keep the comments valid and up-to-date. But Java 17 brings more strict checks for missing Javadocs. So as a result, we have lot of missing Javadoc errors. We decided to proceed with move to Java 17 and for now, we disabled the `failOnWarnings` flag for the `topic-operator` module. But we should get back to it, fix it and re-enable it again.

### PR DESCRIPTION
**Pull request by ShubhamRwt**
Signed-off-by: ShubhamRwt <srawat@redhat.com>

### Type of change

- Refactoring

### Description

_Please describe your pull request_

This PR fixes #7703. This PR fixes the Javadocs issues in the topic-operator module  adding the missing Javadoc or making the methods / fields not public where they don't need to be public.
It also contains refactoring of some methods which are now not used.

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]
Assuming the tests pass, it is OK I guess. @tombentley as a TO SME would need to comments on the detailed changes.

File: topic-operator/src/main/java/io/strimzi/operator/topic/ConflictingChangesException.java [hunk 1]
```diff
@@ -11,6 +11,12 @@
  * and the topic in Kafka had conflicting changes.
  */
 public class ConflictingChangesException extends OperatorException {
+
+    /** Constructor
```
Comment (tombentley) [inline topic-operator/src/main/java/io/strimzi/operator/topic/ConflictingChangesException.java:15]: Eugh, can we put this on the following line? (And same for other instances, you might be able to fix this with a suitable `sed` invocation).

**Review by tombentley — COMMENTED** [review]
This looks fine to me except the non-standard `/** Blah blah` lines where the doc is not just a one-liner (i.e. where the line doesn't end `*/`).

File: topic-operator/src/main/java/io/strimzi/operator/topic/ConflictingChangesException.java (hunk 1 above)
Comment (ShubhamRwt) [reply topic-operator/src/main/java/io/strimzi/operator/topic/ConflictingChangesException.java]: Fixed @tombentley

**Review by tombentley — APPROVED** [review]
Thanks @ShubhamRwt

**Pull request merged**